state/logs.go

```diff
@@ -760,6 +765,14 @@ func (t *logTailer) paramsToSelector(params LogTailerParams, prefix string) bson
 		sel = append(sel,
 			bson.DocElem{"m", bson.M{"$not": bson.RegEx{Pattern: makeModulePattern(params.ExcludeModule)}}})
 	}
+	if len(params.IncludeLabel) > 0 {
+		sel = append(sel,
+			bson.DocElem{"c", bson.M{"$in": params.IncludeLabel}})
```

Comment:
Do we need some kind of index for this query (and the $nin one)?

Reply:
I 100% bet we don't have any indexes on this stuff. I'll check, might be a follow-up PR.

Reply:
https://github.com/juju/juju/pull/13160